integration/e2e/chaincode_service_test.go

```diff
@@ -0,0 +1,275 @@
+				Name:          "TwoOrgsBaseProfileChannel",
+				Consortium:    "SampleConsortium",
+				Orderers:      []string{"orderer"},
+				Organizations: []string{"Org1", "Org2"},
+			})
+			network.Channels = append(network.Channels, &nwo.Channel{
+				Name:        "baseprofilechannel",
+				Profile:     "TwoOrgsBaseProfileChannel",
+				BaseProfile: "TwoOrgsOrdererGenesis",
+			})
+
+			network.GenerateConfigTree()
+
+			// package connection.json
+			extcc.CodeFiles = map[string]string{
+				chaincodeConnectionPath(testDir, extcc.Name): "connection.json",
+			}
+
+			for _, peer := range network.PeersWithChannel("testchannel") {
+				core := network.ReadPeerConfig(peer)
+				core.VM = nil
+				network.WritePeerConfig(peer, core)
+			}
+			network.Bootstrap()
+
+			networkRunner := network.NetworkGroupRunner()
+			process = ifrit.Invoke(networkRunner)
+			Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+		})
+
+		AfterEach(func() {
+			if datagramReader != nil {
+				datagramReader.Close()
+			}
+			if ccserver != nil {
+				ccserver.Signal(syscall.SIGTERM)
+				Eventually(ccserver.Wait(), network.EventuallyTimeout).Should(Receive())
+			}
+		})
+
+		It("executes a basic solo network with 2 orgs and external chaincode service", func() {
+			By("getting the orderer by name")
+			orderer := network.Orderer("orderer")
+
+			By("setting up the channel")
+			network.CreateAndJoinChannel(orderer, "testchannel")
+			nwo.EnableCapabilities(network, "testchannel", "Application", "V2_0", orderer, network.Peer("Org1", "peer0"), network.Peer("Org2", "peer0"))
+
+			By("deploying the chaincode")
+			nwo.DeployChaincodeWithoutInitialization(network, "testchannel", orderer, extcc)
+
+			By("starting the chaincode service")
+			extcc.SetPackageIDFromPackageFile()
+
+			// start external chain code service
+			ccrunner := chaincodeServerRunner(extcc.Path, extcc.PackageID, append([]string{extcc.PackageID, chaincodeServerAddrress}, certFiles...))
+			ccserver = ifrit.Invoke(ccrunner)
+			Eventually(ccserver.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+			// init the chaincode, if required
+			if extcc.InitRequired {
+				By("initing the chaincode")
+				nwo.InitChaincode(network, "testchannel", orderer, extcc, network.PeersWithChannel("testchannel")...)
+			}
+
+			By("getting the client peer by name")
+			peer := network.Peer("Org1", "peer0")
+
+			RunQueryInvokeQuery(network, orderer, peer, "testchannel")
+			RunRespondWith(network, orderer, peer, "testchannel")
+		})
+	})
+})
+
+func chaincodeCertsDir(tempDir string, chaincodeID string) string {
+	return filepath.Join(tempDir, "certs", chaincodeID)
+}
+
+func chaincodeConnectionDir(tempDir string, chaincodeID string) string {
+	return filepath.Join(tempDir, "chaincode-connections", chaincodeID)
+}
+
+func chaincodeConnectionPath(tempDir string, chaincodeID string) string {
+	return filepath.Join(chaincodeConnectionDir(tempDir, chaincodeID), "connection.json")
+}
```

Comment:
Why functions for these instead of vars in the test setup?  They should be calculated once.